spring-boot-project/spring-boot/src/main/java/org/springframework/boot/env/RandomValuePropertySource.java

```diff
@@ -136,9 +136,9 @@ private void assertPresent(boolean present, Range<?> range) {
 	}
 
 	private Object getRandomBytes() {
-		byte[] bytes = new byte[32];
+		byte[] bytes = new byte[128 / 8];
 		getSource().nextBytes(bytes);
-		return DigestUtils.md5DigestAsHex(bytes);
+		return HexFormat.of().withLowerCase().formatHex(bytes);
```

Comment:
Before we get 32 random bytes, and then used MD5 on them to get a hex string. I removed the MD5, we now get 128 bits (output size of MD5) of random bytes directly.